client/src/components/Feed/Post.js

```diff
@@ -33,12 +37,20 @@ const Post = ({
   postDelete,
   user,
 }) => {
-  const page = useRef(1);
+  const page = useRef(0);
+  const limit = useRef(5);
+  const loadedComments = useRef(0);
   const { postId } = useParams();
+  const history = useHistory();
   const [showComments, setShowComments] = useState(false);
   const [copied, setCopied] = useState(false);
-  const [currentPost, setCurrentPost] = useState({});
+  const [firstLoad, setFirstLoad] = useState(true);
+  const [loading, setLoading] = useState(false)
+  const [loadVisibility, setLoadVisibility] = useState(true)
   const [modalVisibility, setModalVisibility] = useState(false);
+  const [comment, setComment] = useState([]);
```

Comment:
looks like a lot of state here, probably better to replace multiple state hooks with `useReducer`

Reply:
thanks!